mysql_tests/test_crud.py

```diff
@@ -0,0 +1,333 @@
+
+    db2 = gino.Gino(MYSQL_URL)
+    await db2.set_bind(MYSQL_URL, autocommit=True)
+
+    class NameCard(db2.Model):
+        __tablename__ = "name_cards"
+
+        last_name = db2.Column(db2.Unicode(255), primary_key=True)
+        first_name = db2.Column(db2.Unicode(255), primary_key=True)
+
+    await db2.gino.create_all()
+
+    try:
+        await NameCard.create(first_name="first", last_name="last")
+        nc = await NameCard.get(("last", "first"))
+        assert nc.first_name == "first"
+        assert nc.last_name == "last"
+        nc = await NameCard.get(dict(first_name="first", last_name="last"))
+        assert nc.first_name == "first"
+        assert nc.last_name == "last"
+        nc = await NameCard.get({1: "first", "last_name": "last"})
+        assert nc.first_name == "first"
+        assert nc.last_name == "last"
+    finally:
+        await db2.gino.drop_all()
+        await db2.pop_bind().close()
+
+
+async def test_connection_as_bind(engine):
+    async with engine.acquire() as conn:
+        await test_get(conn)
+
+
+async def test_update(engine, random_name):
+    u1 = await test_create(engine)
+    await u1.update(nickname=random_name).apply(bind=engine, timeout=10)
+    u2 = await User.get(u1.id, bind=engine)
+    assert u2.nickname == random_name
+
+
+async def test_update_missing(engine, random_name):
+    from gino.exceptions import NoSuchRowError
+
+    u1 = await test_create(engine)
+    rq = u1.update(nickname=random_name)
+    await u1.delete(bind=engine)
+    with pytest.raises(NoSuchRowError):
+        await rq.apply(bind=engine, timeout=10)
+
+
+async def test_update_multiple_primary_key(engine):
+    u1 = await test_create(engine)
+    u2 = await test_create(engine)
+    u3 = await test_create(engine)
+    await Friendship.create(bind=engine, my_id=u1.id, friend_id=u2.id)
+    f = await Friendship.get((u1.id, u2.id), bind=engine)
+    await f.update(my_id=u2.id, friend_id=u3.id).apply(bind=engine)
+    f2 = await Friendship.get((u2.id, u3.id), bind=engine)
+    assert f2
+
+
+async def test_delete(engine):
+    u1 = await test_create(engine)
+    await u1.delete(bind=engine, timeout=10)
+    u2 = await User.get(u1.id, bind=engine)
+    assert not u2
+
+
+async def test_delete_bind(bind):
+    u1 = await test_create(bind)
+    await u1.delete(timeout=10)
+    u2 = await User.get(u1.id)
+    assert not u2
+
+
+async def test_delete_multiple_primary_key(engine):
+    u1 = await test_create(engine)
+    u2 = await test_create(engine)
+    f = await Friendship.create(bind=engine, my_id=u1.id, friend_id=u2.id)
+    await f.delete(bind=engine)
+    f2 = await Friendship.get((u1.id, u2.id), bind=engine)
+    assert not f2
+
+
+async def test_string_primary_key(engine):
+    relations = ["Colleagues", "Friends", "Lovers"]
+    for r in relations:
+        await Relation.create(bind=engine, timeout=10, name=r)
+    r1 = await Relation.get(relations[0], bind=engine, timeout=10)
+    assert r1.name == relations[0]
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/gh/python-gino/gino/pullRequest?prid=5587950)